BranchMetrics/ios-branch-deep-linking-attribution
# (WIP) - SDK 2223 DMA Consent Params Implementation

## Reference
SDK-2223 -- Design and implement key/values to represent consent param

## Summary
* Added following API in Branch Class.
` + (void) setDMAParamsForEEA:(BOOL) eeaRegion AdPersonalizationConsent:(BOOL) adPersonalizationConsent AdUserDataUsageConsent:(BOOL) adUserDataUsageConsent;
`
* Branch SDK will save DMA Compliance Parameters, set by developer using above API, and will send to server with install, open and event requests. BranchPreferenceHelper class is updated to store and read these params.
* Added unit test for testing above API

## Motivation
Support new params added in Google Conversion API for DMA Compliance.

## Type Of Change

## Testing Instructions
* All unit tests should pass.
* Call API setDMAParamsForEEA and check 
         -- SDK is sending value of `eeaRegion`, with every `/v1/install` , `/v1/open`, `/v2/event` , `v1/pageview` request.
         -- SDKis sending value of `adPersonalizationConsent` / `adUserDataUsageConsent` with every `/v1/install` , `/v1/open`, `/v2/event` , `v1/pageview` request only if -
                      - `eeaRegion` is `true` **OR** 
                      - `eeaRegion` is `false` and `adPersonalizationConsent` / `adUserDataUsageConsent` is also set to `false`
            Ref : https://www.notion.so/brancheng/Google-DMA-Compliance-c19ed28d2ea74b2b9fa4d2d77f8cdcbe 

cc @BranchMetrics/saas-sdk-devs for visibility.
